kotest-framework/kotest-framework-engine/src/jvmTest/kotlin/com/sksamuel/kotest/engine/threads/BeforeTestConcurrentInstancePerLeafTest.kt

```diff
@@ -22,7 +21,7 @@ class SpecThreadBeforeTestConcurrentInstancePerLeafTest : FunSpec({
       if (isLockAcquired) {
          lock.lock()
          try {
-            delay(300)
+            Thread.sleep(300)
```

Comment:
How come this was changed ?

Reply:
Because the thread is still allocated so the executor is unable to reuse it ?

Reply:
Seems like that. I haven't looked into the Executor's internals. Might be they use a spin lock and re-use the same thread if just a short time period elapses. If it takes longer or a thread is actually parked, they'd use another thread. But that's all implementation-dependent and could change. The comment for `Executor.execute()` says
> Executes the given command at some time in the future. The command may execute in a new thread, in a pooled thread, or in the calling thread, at the discretion of the Executor implementation.

Reply:
Could also write our own executor implementation that always creates new threads for the purposes of testing.

Reply:
Sure, I also have more thought on that. Probably too much for this iteration, but briefly: Testing as close to common practice (Kotlin default/promoted ways of using concurrency) could be the default in Kotest. Then we should accept that coroutines will always hop threads, even if parallel execution limits may be in effect. We should also cease to rely on thread guarantees, since we deal with coroutines, not threads.

Then, of course, we could offer other mechanisms for special purposes, e.g. a dispatcher with a strictly round-robin thread pool. But we should be clear about what our goals are when asking for special-purpose dispatchers.

Reply:
I'm happy with the thread.sleep as its in improvement on delay.